### DISCUSSION TITLE
How to send data via MIDI?

### DISCUSSION THREAD

**Discussion by dcoredump** [probonopd/MiniDexed#245] [Q&A] [answered]
Hi (@probonopd @rsta2)

I want to send a voice dump via MIDI. My problem is that I am not sure how to do this. Currently I tried to copy the code from MIDIThru sending but I must say that I don't fully understand what there is happening. What is the easiest way to send a MIDI message by using `CSerialMIDIDevice::Send()`?

My current code is [here](https://github.com/dcoredump/MiniDexed). The dump method is at the end of `mididevices.cpp`  (`CMIDIDevice::SendSystemExclusiveVoice()`).

TIA, Holger

**Comment by rsta2** [comment, marked as answer]
@dcoredump You just can call `Send(pMessage, nLength, nCable)` from `CMIDIDevice`. That `DeviceMap` and `Iterator` stuff is only needed for MIDI Thru, but not for sending to a MIDI Device, you already have a pointer for.

**Reply by dcoredump** [reply]
Ok, got it now! Thanks!!!

**Comment by probonopd** [comment]
@dcoredump Edisyn has a "Request Current Patch" (Ctrl-R) menu item. It would be neat if we could support that. I think "Request Current Patch" sends a sysex command to the instrument and then expects the instrument to send back the voice sysex.

**Reply by dcoredump** [reply]
Yes, that's what I am trying. I am also ready that it could work now, but unfortunately my USB-MIDI adapters under Linux and WIndows make permanent trouble, so that I can't test properly again.

With [Herr Mueller Online](https://dx7.vstforx.de/) you can actually test (it sends a complete voice dump for every small change, but this is exactly what I want to try - and it recognizes voice dumps and shows all parameters), but my USB-MIDI adapters are not recognized by the browser.

With MicroDexed I did it also in such a way that with the selection of a new voice a SYSEX-voice-dump is sent. Then you can see the current settings in the editor. But it doesn't work yet, because I don't know how to call a method from `CMIDIDevice` in `CMiniDexed::ProgramChange()`.

**Reply by probonopd** [reply]
> With MicroDexed I did it also in such a way that with the selection of a new voice a SYSEX-voice-dump is sent. 

Smart!
